resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackendSuite.scala

```diff
@@ -0,0 +1,383 @@
+/*
```

Comment:
Huge thanks for this. This test one of most important additions to keep the cluster backend healthy, correct and maintainable.